### PR TITLE
Improved pangram algorithm using regular expressions

### PR DESCRIPTION
**Pull request by fahimfaisaal**
# Welcome to the JavaScript community

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

* [x] upgrade the **Pangram** algorithm.
* [x] add Js doc.

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new JavaScript files are placed inside an existing directory.
* [x] All filenames should use the UpperCamelCase (PascalCase) style.  There should be no spaces in filenames.

### CODE REVIEW

File: String/CheckPangram.js [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-  return frequency.size === 26
+  /**
+   * match all 26 alphabets using regex, with the help of:
+   * Capturing Group
+   * Character set
+   * Negative lookahead
+   * Dot & star
+   * Numeric reference
+ */
+  return string.match(/([a-z])(?!.*\1)/gi).length === 26
```
Comment (raklaptudirm) [inline String/CheckPangram.js:20]: Please explain your regex in detail as it will seem hard to beginners.

**Review by raklaptudirm — APPROVED** [review]

File: String/CheckPangram.js (hunk 1 above)
Comment (fahimfaisaal) [reply String/CheckPangram.js:20]: Ok, 🙂

Comment (raklaptudirm) [reply String/CheckPangram.js:20]: You can make them two different functions.

Comment (fahimfaisaal) [reply String/CheckPangram.js:23]: Oke

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**